osu.Framework/Testing/TestScene.cs

```diff
@@ -37,28 +39,69 @@ public abstract class TestScene : Container, IDynamicallyCompile
         private Task runTask;
         private ITestSceneTestRunner runner;
 
+        /// <summary>
+        /// A nested game instance, if added via <see cref="AddGame"/>.
+        /// </summary>
+        private Game nestedGame;
+
         public object DynamicCompilationOriginal { get; internal set; }
 
         [OneTimeSetUp]
         public void SetupGameHost()
```

Comment:
Although slightly out of scope from this PR, I think renaming this method to something like `SetupHeadless` would be better as to not confuse it of being performed in all scenarios (which now, with `TestSceneHost` instead of `TestRunHeadlessGameHost`, makes it think more like so)

(also same would apply to `DestroyGameHost`)
```suggestion
        public void SetupHeadless()
```

Reply:
I've renamed in line with the other nunit specific methods.